### ISSUE TITLE
Error with container: 'grafana.ini' is not readable.                                                      

### ISSUE THREAD

**Issue by antonywinn** [livehouse-automation/livehouse-docker-grafana#5]
Newly created amd64 container on QNAP

using shared volumes as below:

```
/var/lib/grafana | /share/Containers/grafana/dashboards
/etc/grafana | /share/Containers/grafana/config
/var/lib/grafana-plugins | /share/Containers/grafana/plugins
```

All shared volumes empty on startup.

Errors during startup:

```
GF_PATHS_CONFIG='/etc/grafana/grafana.ini' is not readable.                                                                                                  
You may have issues with file permissions, more information here: http://docs.grafana.org/installation/docker/#migration-from-a-previous-version-of-the-docke
r-container-to-5-1-or-later                                                                                                                                  
CRIT[07-30|04:24:28] Failed to parse /etc/grafana/grafana.ini, open /etc/grafana/grafana.ini: no such file or directory%!(EXTRA []interface {}=[])  
```

Added a default grafana.ini to /share/Containers/grafana/config and container started, but there were no installed plugins/panels. 

At a guess, looks like container first run needs to put a grafana.ini file in the external folder if it's mapped, and that ini file should contain any customisations for panels/datasources/default dashboards.

**Comment by mikenye** [comment]
As a workaround, if you can, change the ownership of grafana.ini to be owned by user ID 472.

eg: "chown 472 /share/Containers/grafana/config/grafana.ini" from the CLI of your NAS.

**Comment by antonywinn** [comment]
It's reading the file fine since I put it in there, so not a permissions issue (unless you mean the container should have created the file and failed)?

I think it's getting upset about having the /etc folder mapped from the internal one in the container which does contain a valid grafana.ini file.

**Comment by mikenye** [comment]
Newer builds of Grafana are finnicky when it comes to permissions. It wants its files owned by the grafana user (472).

I've added some code to change ownership on container start to make this an easy fix, just testing then I'll update the docker image.

**Comment by antonywinn** [comment]
I may be missing something - but how will it change the owner on the file if it doesn't exist? I had to create and put a file into the folder. The container didn't create a grafana.ini file in the shared folder.

**Comment by mikenye** [comment]
Its probably trying to create the file (as user ID 472, which probably doesn't exist on your NAS) and then failing. If you don't need to manually put stuff in grafana.ini, suggest leaving it unmapped.

You should only need to map /var/lib/grafana (so your dashboards & stuff are saved), all of the config you should need to do can be done via environment variables.

**Comment by antonywinn** [comment]
Ok, makes sense to leave it unmapped then. Let me know if you want me to test the updated version as I have to recreate the container to remove the mapping.

**Comment by mikenye** [comment]
Nope. I'm going to hold off on updating the container. I misread your original issue and thought it was having trouble reading a grafana.ini that you'd created yourself. If the container doesn't have permissions to *create* the file, that's different. :)

**Comment by mikenye** [comment]
Let me know how you fare with the config dir unmapped. If its OK then I'll just update the readme letting people know about the permissions for the directory containing grafana.ini if they want it mapped.

**Comment by antonywinn** [comment]
Same error - using its own internal location for the /etc folder.

```

/var/lib/grafana | /share/Containers/grafana/dashboards | Read/Write
-- | -- | --
/var/lib/grafana-plugins | /share/Containers/grafana/plugins | Read/Write
/etc/grafana | /share/CACHEDEV6_DATA/Containers/container-station-data/lib/docker/volumes/ed85fac67080f4b3f6d4936a9907e1e141a037b7d715c81215f96649875c49af/_data | Read/Write


```

**Comment by mikenye** [comment]
I've fixed the issue (hopefully), and pushed a new docker image. 

Please try livehouseautomation/veraflux-grafana:5.1.x-amd64 and let me know if this fixes the problem.

If so I'll re-build the arm32v7 image and re-do the manifest for "latest".

**Comment by mikenye** [comment]
Fixed and tested in commit 17cb17c90d27fa9a9f83d2e73a66b645bad728a6.

Docker images updated. You should now be able to pull the latest on arm32v7 or amd64.
